packages/devtools_app/lib/src/debugger/program_explorer_controller.dart

```diff
@@ -77,15 +77,11 @@ class ProgramExplorerController extends DisposableController
       this,
       libraries,
     );
-    _rootObjectNodes.addAll(nodes);
+    _rootObjectNodes.replaceAll(nodes);
     _initialized.value = true;
   }
 
   void initListeners() {
-    addAutoDisposeListener(
```

Comment:
This was removed because it was redundant - the `debugger_controller` was also listening for `serviceManager.isolateManager.selectedIsolate` [code link](https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/debugger/debugger_controller.dart#L123-L125), and was re-populating the scripts when the isolate changed. Therefore, what was happening was:

1. isolate changes -> refresh
2. sorted scripts changes -> refresh

Since we only want to refresh once we have the scripts, this first listener could be removed.